src/ucp/api/ucp.h

```diff
@@ -517,10 +517,13 @@ typedef enum {
 typedef enum {
     UCP_OP_ATTR_FIELD_REQUEST       = UCS_BIT(0),  /**< request field */
     UCP_OP_ATTR_FIELD_CALLBACK      = UCS_BIT(1),  /**< cb field */
-    UCP_OP_ATTR_FIELD_DATATYPE      = UCS_BIT(2),  /**< datatype field */
+    UCP_OP_ATTR_FIELD_USER_DATA     = UCS_BIT(2),  /**< user_data field */
+    UCP_OP_ATTR_FIELD_DATATYPE      = UCS_BIT(3),  /**< datatype field */
 
     UCP_OP_ATTR_FLAG_NO_IMM_CMPL    = UCS_BIT(16), /**< deny immediate completion */
-    UCP_OP_ATTR_FLAG_NO_ZCOPY       = UCS_BIT(17), /**< do not use zcopy proto */
+    UCP_OP_ATTR_FLAG_FAST_CMPL      = UCS_BIT(17), /**< expedite local completion,
```

Comment:
What does this really mean ?

Reply:
Seems like you switch to copy-in/out protocol instead of zero copy. Does it expedite in call cases ? You can run out of buffers, delay the operation all together, etc.

Reply:
in currently implementation, it disables zero-copy and rendezvous, for blocking calls like MPI_Send where we need the request to complete as soon as possible.

Reply:
So, seems like previous name reflects better what it does , or I'm missing something  ?  Maybe add: `this option disables zero copy randevouze protocol and forces the library to use copy-in/out protocol`

Reply:
> So, seems like previous name reflects better what it does , or I'm missing something ? Maybe add: `this option disables zero copy randevouze protocol and forces the library to use copy-in/out protocol`

The previous name reflects what it does now, but IMHO we should keep API names independent of implementation. copy/zcopy is not part of the UCP API. We can document "For example, this option can disable zero copy and/or rendezvous protocols which require synchronization with the remote peer before releasing the local send buffer", however it's more of an "advice for implementer" rather than API specification.

Reply:
I like your suggestion. Let's start introducing "A note for implementer:"

Reply:
added note